### DISCUSSION TITLE
The Approach to Rooms

### DISCUSSION THREAD

**Discussion by thewane** [zibonbadi/liquidms#12] [General]
Every Master Server will naturally have its own rooms/server lists. With that being said, do we differentiate between rooms that have conflicting numbers (example: MS1 has room 27, and so does MS2), and what would that look like?

**Comment by zibonbadi** [comment]
An idea may be to differentiate rooms by origin and auto-generate room names. Example:

- `default@mb.srb2.org`
- `casual@alice.liquidms.dummy`
- `kart@bob.liquidms.dummy`

Combined with my approach mentioned in #5 (2021-09-29), this could lead to a cleanly organized, if slow to update system based on distributed redundancy.

**Comment by thewane** [comment]
That sounds good to me. So basically we generate our own rooms files instead of using any specific master servers by requesting the rooms from every MS loaded every once in a while (cron jobs, as I said in issue #5), and parsing said files manually (how this will be implemented can be up for debate, but I personally think above would work good), checking for redundancy as well. This allows for each part to be isolated. The parser, the requester, the server, and the scheduler can all be different files or functions (_or_ programs in the case of the scheduler and/or server), making it much more modular than other approaches.

**Comment by zibonbadi** [comment]
I'm not sure if we should do a cron job or a script that periodically polls the external servers. On one hand, we should follow DOTADIW, but if you're e.g. hosting on Windows for some reason, it may be a good idea to have a fetch script that can be used as a cronjob with a simple daemon script that calls the former.
